pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -303,6 +312,118 @@ func (l *LibvirtDomainManager) setGuestTime(vmi *v1.VirtualMachineInstance) erro
 	return nil
 }
 
+// Set interfaces link down and up to renew DHCP leases
+func (l *LibvirtDomainManager) reconnectGuestNics(vmi *v1.VirtualMachineInstance) error {
+	logger := log.Log.Object(vmi)
+	domName := api.VMINamespaceKeyFunc(vmi)
+	dom, err := l.virConn.LookupDomainByName(domName)
+	if err != nil {
+		log.Log.Object(vmi).Reason(err).Error(failedReconnectGuestNic)
+		return err
+	}
+	defer dom.Free()
+	xmlstr, err := dom.GetXMLDesc(0)
+	if err != nil {
+		return err
+	}
+
+	var domain api.DomainSpec
+	err = xml.Unmarshal([]byte(xmlstr), &domain)
+	if err != nil {
+		return fmt.Errorf("parsing domain XML failed, err: %v", err)
+	}
+
+	// Look up all the interfaces and reconnect them
+	for _, iface := range domain.Devices.Interfaces {
+		specIface := getIfaceByName(vmi, iface.Alias.GetName())
+
+		// For masquerade and slirp we don't have to do anything
+		if specIface.Masquerade != nil || specIface.Slirp != nil {
```

Comment:
Your work is only focused on bridged interfaces, so would it make more sense to only check whether the interface is bridged or not?
```
if specIface.Bridge == nil {
      continue
}
```

Reply:
I'm considering that https://github.com/kubevirt/kubevirt/pull/7648 will be merged next to this one

Reply:
Thant makes sense, thank you